static/gsApp/views/onDemandBudgets/utils.tsx

```diff
@@ -156,35 +147,10 @@ function getBudgetMode(budget: OnDemandBudgets) {
 
 export function getOnDemandBudget(budget: OnDemandBudgets, dataCategory: DataCategory) {
   if (budget.budgetMode === OnDemandBudgetMode.PER_CATEGORY) {
-    // TODO(data categories): check if this can be parsed
-    switch (dataCategory) {
-      case DataCategory.ERRORS: {
-        return budget.budgets.errors ?? 0;
-      }
-      case DataCategory.TRANSACTIONS: {
-        return budget.budgets.transactions ?? 0;
-      }
-      case DataCategory.ATTACHMENTS: {
-        return budget.budgets.attachments ?? 0;
-      }
-      case DataCategory.REPLAYS: {
-        return budget.budgets.replays ?? 0;
-      }
-      case DataCategory.MONITOR_SEATS: {
-        return budget.budgets.monitorSeats ?? 0;
-      }
-      case DataCategory.UPTIME: {
-        return budget.budgets.uptime ?? 0;
-      }
-      case DataCategory.PROFILE_DURATION: {
-        return budget.budgets.profileDuration ?? 0;
-      }
-      case DataCategory.PROFILE_DURATION_UI: {
-        return budget.budgets.profileDurationUI ?? 0;
-      }
-      default:
-        return getTotalBudget(budget);
+    if (dataCategory in budget.budgets) {
+      return budget.budgets[dataCategory] ?? 0;
     }
+    return getTotalBudget(budget);
   }
   return getTotalBudget(budget);
```

Comment:
nit: we can change the `if (budget.budgetMode === OnDemandBudgetMode.PER_CATEGORY)` to `if (budget.budgetMode === OnDemandBudgetMode.PER_CATEGORY && dataCategory in budget.budgets)` so we only need one instance of `return getTotalBudget(budget)`